component/block_cache/block_cache.go

```diff
@@ -350,18 +350,17 @@ func (bc *BlockCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Han
 			return nil, fmt.Errorf("failed to retrieve block list for %s", options.Name)
 		}
 
+		valid := bc.validateBlockList(options.Name, bc.blockSize, blockList)
+		if !valid {
+			return nil, fmt.Errorf("block size mismatch for %s", options.Name)
+		}
+
+		// blockList is valid, Copy the blockList to the handle
 		lst, _ := handle.GetValue("blockList")
```

Comment:
from here till the end of the loop can be moved to another function instead of iterating the blocks twice.